File: episode-51-r/fib3.r
#!/usr/bin/env Rscript

suppressPackageStartupMessages(require(matrixcalc))

fib = function(n) {
  m = matrix(c(1,1,1,0), ncol=2)
  matrix.power(m, n)[1,2]
}

for (i in seq(1, 20)) {
  cat("fib(", i, ") = ", fib(i), "\n", sep="")
}

File: episode-100-ruby/fib3.rb
#!/usr/bin/env ruby

def fib(n)
  @fib ||= {}
  @fib[n] ||= begin
    if n <= 2
      1
    else
      fib(n-1) + fib(n-2)
    end
  end
end

(200..210).each do |n|
  puts "fib(#{n}) = #{fib(n)}"
end
